With `-quickfix` enabled, `[rewritten by quickfix]` gets printed for every error, regardless of whether or not it was actually fixed/fixable

## Reproduction steps

Scala version: 2.13.12

in Test.scala:
```scala
import foo.bar
```
Compiling with `-quickfix:any` prints
```
Test.scala:1: error: [rewritten by -quickfix] not found: object foo
import foo.bar
       ^
1 error
```
## Problem
It says the "not found: object foo" error was "rewritten by -quickfix" but it wasn't, it's not a `quickfixable` error

--- a/src/compiler/scala/tools/nsc/Reporting.scala
+++ b/src/compiler/scala/tools/nsc/Reporting.scala
@@ -86,23 +86,23 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
     private val skipRewriteAction = Set(Action.WarningSummary, Action.InfoSummary, Action.Silent)
 
     private def registerTextEdit(m: Message): Boolean =
-      if (quickfixFilters.exists(f => f.matches(m))) {
-        textEdits.addAll(m.actions.flatMap(_.edits))
+      m.actions.exists(_.edits.nonEmpty) &&
+      quickfixFilters.exists(_.matches(m)) && {
+        m.actions.foreach(action => textEdits.addAll(action.edits))
         true
       }
-      else false
 
-    private def registerErrorTextEdit(pos: Position, msg: String, actions: List[CodeAction]): Boolean = {
-      val matches = quickfixFilters.exists({
+    private def registerErrorTextEdit(pos: Position, msg: String, actions: List[CodeAction]): Boolean =
+      actions.exists(_.edits.nonEmpty) &&
+      quickfixFilters.exists {
         case MessageFilter.Any => true
         case mp: MessageFilter.MessagePattern => mp.check(msg)
         case sp: MessageFilter.SourcePattern => sp.check(pos)
         case _ => false
-      })
-      if (matches)
-        textEdits.addAll(actions.flatMap(_.edits))
-      matches
-    }
+      } && {
+        actions.foreach(action => textEdits.addAll(action.edits))
+        true
+      }
 
     private val summarizedWarnings: mutable.Map[WarningCategory, mutable.LinkedHashMap[Position, Message]] = mutable.HashMap.empty
     private val summarizedInfos: mutable.Map[WarningCategory, mutable.LinkedHashMap[Position, Message]] = mutable.HashMap.empty

--- a/src/compiler/scala/tools/nsc/typechecker/Namers.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/Namers.scala
@@ -1122,7 +1122,7 @@ trait Namers extends MethodSynthesis {
           val leg = legacy.toString
           val help = if (pts != leg) s" instead of $leg" else ""
           val msg = s"under -Xsource:3-cross, the inferred type changes to $pts$help"
-          val src = currentUnit.source
+          val src = tree.pos.source
           val pos = {
             val eql = src.indexWhere(_ == '=', start = tree.rhs.pos.start, step = -1)
             val declEnd = src.indexWhere(!_.isWhitespace, start = eql - 1, step = -1) + 1

--- a/test/junit/scala/tools/nsc/QuickfixTest.scala
+++ b/test/junit/scala/tools/nsc/QuickfixTest.scala
@@ -10,8 +10,10 @@ import scala.tools.testkit.BytecodeTesting
 import scala.tools.testkit.ReleasablePath._
 import scala.util.Using
 
+import reporters.StoreReporter
+
 class QuickfixTest extends BytecodeTesting {
-  def testQuickfix(a: String, b: String, args: String): Unit = if (!scala.util.Properties.isWin) {
+  def testQuickfix(a: String, b: String, args: String, checkInfo: StoreReporter.Info => Boolean = _ => true): Unit = if (!scala.util.Properties.isWin) {
     Using.resource(Files.createTempFile("unitSource", "scala")) { src =>
       Files.write(src, a.getBytes)
       val c = BytecodeTesting.newCompiler(extraArgs = args)
@@ -19,6 +21,8 @@ class QuickfixTest extends BytecodeTesting {
       val f = AbstractFile.getFile(src.toFile.getAbsolutePath)
       r.compileSources(List(new BatchSourceFile(f)))
       assertEquals(b, new String(Files.readAllBytes(src)))
+      for (info <- c.global.reporter.asInstanceOf[StoreReporter].infos)
+        assert(checkInfo(info), info)
     }
   }
 
@@ -80,5 +84,10 @@ class QuickfixTest extends BytecodeTesting {
     testQuickfix(b, c, "-deprecation -quickfix:msg=Auto-application")
     testQuickfix(c, c, "-quickfix:cat=deprecation") // without -deprecation, the warning is not shown and not rewritten
     testQuickfix(c, d, "-deprecation -quickfix:cat=deprecation")
+  }
+
+  @Test def `do not lie about fixing`: Unit = {
+    val a = "import foo.bar"
+    testQuickfix(a, a, "-quickfix:any", !_.msg.contains("[rewritten by -quickfix]"))
   }
 }

